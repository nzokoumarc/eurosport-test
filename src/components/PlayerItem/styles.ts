import { makeStyles, Theme } from '@material-ui/core';

export const useStyles = makeStyles((theme: Theme) => ({
  root: {
    margin: theme.spacing(2, 3),
    position: 'relative',
  },
  flex: {
    display: 'flex',
  },
  link: {
    position: 'absolute',
    top: 0,
    width: '100%',
    height: '100%',
    border: '1px solid rgba(0,0,0,0.14)',
    '&:hover': {
      boxShadow:
        '0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12)',
    },
  },
  color: {
    color: theme.palette.grey[500],
  },
  content: {
    width: '100%',
    alignItems: 'center',
    padding: theme.spacing(2),
  },
}));
