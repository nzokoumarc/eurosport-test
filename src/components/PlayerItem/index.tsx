import * as React from 'react';
// css framework
import { Typography } from '@material-ui/core';
// type of datas
import { PlayersType } from '../../utils/players';
// component
import PlayerStats from '../PlayerStats';

import { useStyles } from './styles';

// type props
type PlayerItemProps = {
  item: PlayersType;
};

const PlayerItem: React.FC<PlayerItemProps> = React.memo(({ item }) => {
  const classes = useStyles('');
  return (
    <>
      <div className={classes.root}>
        <div className={classes.flex}>
          <img src={item.picture} alt={item.shortname} width="150" />
          <div className={classes.content}>
            <div>
              <Typography color="primary" variant="h5">
                {item.firstname} {item.lastname}
              </Typography>
              <Typography color="initial" variant="body2">
                <span className={classes.color}>Sex : </span>
                {item.sex}
              </Typography>
              <Typography color="initial" variant="body2">
                <span className={classes.color}>Country : </span>
                <img src={item.country.picture} alt={item.country.code} width="20" />
                &nbsp; {item.country.code}
              </Typography>
              <PlayerStats data={item.data} />
            </div>
          </div>
        </div>
      </div>
    </>
  );
});

export default PlayerItem;
