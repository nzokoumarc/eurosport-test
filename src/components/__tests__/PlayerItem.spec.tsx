import * as React from 'react';
import { shallow } from 'enzyme';

import Mocks from '../__mocks__';
import PlayerItem from '../PlayerItem';

describe('PlayerItem', () => {
  it('renders', () => {
    const component = shallow(<PlayerItem item={Mocks} />);
    expect(component).toMatchSnapshot();
  });
});
