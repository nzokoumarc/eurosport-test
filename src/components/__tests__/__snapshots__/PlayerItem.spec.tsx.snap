// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerItem renders 1`] = `
<Fragment>
  <div
    className="makeStyles-root-1"
  >
    <div
      className="makeStyles-flex-2"
    >
      <img
        alt="S.WAW"
        src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
        width="150"
      />
      <div
        className="makeStyles-content-5"
      >
        <div>
          <WithStyles(ForwardRef(Typography))
            color="primary"
            variant="h5"
          >
            Stan
             
            Wawrinka
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="initial"
            variant="body2"
          >
            <span
              className="makeStyles-color-4"
            >
              Sex : 
            </span>
            M
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="initial"
            variant="body2"
          >
            <span
              className="makeStyles-color-4"
            >
              Country : 
            </span>
            <img
              alt="SUI"
              src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
              width="20"
            />
            Â  
            SUI
          </WithStyles(ForwardRef(Typography))>
          <Memo()
            data={
              Object {
                "age": 33,
                "height": 183,
                "last": Array [
                  1,
                  1,
                  1,
                  0,
                  1,
                ],
                "points": 1784,
                "rank": 21,
                "weight": 81000,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
